compiler/rustc_parse/src/parser/attr.rs

```diff
@@ -416,9 +416,20 @@ impl<'a> Parser<'a> {
             Err(err) => err.cancel(),
         }
 
-        Err(self
-            .dcx()
-            .create_err(InvalidMetaItem { span: self.token.span, token: self.token.clone() }))
+        let token = self.token.clone();
+        let (sugg, or_ident) = if self.prev_token == token::Eq && !self.token.span.from_expansion()
```

Comment:
Please add a comment about what this check does, exactly, and why we only accept identifiers in the `else` (I'm also pretty sure identifiers are unrelated to expansions?...)

Reply:
Added a comment explaining the check, including the `from_expansion()`.

> why we only accept identifiers in the `else`

From briefly glancing over attributes-related tests, it seemed like the "or identifier" would be valid for the remaining cases covered by the else branch, e.g. https://github.com/rust-lang/rust/blob/1a47f5b448f1e023e7ffd2eb57d473d619d2c04d/tests/ui/attributes/nonterminal-expansion.rs#L6

@rustbot review